### PR TITLE
clojure-lsp: 2022.01.22 -> 2022.02.01

### LINKED ISSUE #157574: clojure-lsp: graalvm build error
### Describe the bug

Recently, clojure-lsp build started to fail on unstable and master

### Steps To Reproduce

- check out `nixos-unstable` or `master`
- `nix-build . -A clojure-lsp`

### Screenshots
```
Fatal error: com.oracle.svm.core.util.VMError$HostedError: Option name "AddAllCharsets" has multiple definitions: com.oracle.svm.hosted.jdk.localization.LocalizationFeature$Options.AddAllCharsets and com.oracle.svm.core.jdk.localization.LocalizationFeature$Options.AddAllCharsets
        at com.oracle.svm.core.util.VMError.shouldNotReachHere(VMError.java:68)
        at com.oracle.svm.hosted.option.HostedOptionParser.lambda$collectOptions$0(HostedOptionParser.java:74)
        at com.oracle.svm.common.option.CommonOptionParser.collectOptions(CommonOptionParser.java:172)
        at com.oracle.svm.core.option.SubstrateOptionsParser.collectOptions(SubstrateOptionsParser.java:120)
        at com.oracle.svm.hosted.option.HostedOptionParser.collectOptions(HostedOptionParser.java:63)
        at com.oracle.svm.hosted.option.HostedOptionParser.<init>(HostedOptionParser.java:58)
        at com.oracle.svm.hosted.AbstractNativeImageClassLoaderSupport.setupHostedOptionParser(AbstractNativeImageClassLoaderSupport.java:118)
        at com.oracle.svm.hosted.NativeImageGeneratorRunner.installNativeImageClassLoader(NativeImageGeneratorRunner.java:179)
        at com.oracle.svm.hosted.NativeImageGeneratorRunner.main(NativeImageGeneratorRunner.java:121)
        at com.oracle.svm.hosted.NativeImageGeneratorRunner$JDK9Plus.main(NativeImageGeneratorRunner.java:617)
Error: Image build request failed with exit status 1
com.oracle.svm.driver.NativeImage$NativeImageError: Image build request failed with exit status 1
        at com.oracle.svm.driver.NativeImage.showError(NativeImage.java:1770)
        at com.oracle.svm.driver.NativeImage.build(NativeImage.java:1477)
        at com.oracle.svm.driver.NativeImage.performBuild(NativeImage.java:1438)
        at com.oracle.svm.driver.NativeImage.main(NativeImage.java:1425)
error: builder for '/nix/store/hhqgbqlpiaxq4bsj6jl3lw4yp5mnpagl-clojure-lsp-2022.01.22-01.31.09.drv' failed with exit code 1;
       last 10 log lines:
       >        at com.oracle.svm.hosted.AbstractNativeImageClassLoaderSupport.setupHostedOptionParser(AbstractNativeImageClassLoaderSupport.java:118)
       >         at com.oracle.svm.hosted.NativeImageGeneratorRunner.installNativeImageClassLoader(NativeImageGeneratorRunner.java:179)
       >         at com.oracle.svm.hosted.NativeImageGeneratorRunner.main(NativeImageGeneratorRunner.java:121)
       >  at com.oracle.svm.hosted.NativeImageGeneratorRunner$JDK9Plus.main(NativeImageGeneratorRunner.java:617)
       > Error: Image build request failed with exit status 1
       > com.oracle.svm.driver.NativeImage$NativeImageError: Image build request failed with exit status 1
       >        at com.oracle.svm.driver.NativeImage.showError(NativeImage.java:1770)
       >  at com.oracle.svm.driver.NativeImage.build(NativeImage.java:1477)
       >      at com.oracle.svm.driver.NativeImage.performBuild(NativeImage.java:1438)
       >       at com.oracle.svm.driver.NativeImage.main(NativeImage.java:1425)

```

### Additional context

Bisecting suggests, that this is because of a recent update to graalvm -> 22.0.0.2

https://github.com/NixOS/nixpkgs/commit/a362925794d7b6f7e562b4c49136ac685d1e38cc

#### Attempted variations

- graalvm17
- changing command line parameters to match https://github.com/clojure-lsp/clojure-lsp/blob/master/cli/graalvm/native-unix-compile.sh
- updating clojure-lsp (already is at latest version)

### Notify maintainers

@ericdallo @babariviere 

### Metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.93, NixOS, 22.05 (Quokka)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.5.1`
 - channels(root): `""`
 - channels(herwig): `""`
 - nixpkgs: `/etc/nixpkgs`

### PR DESCRIPTION
**Pull request by willcohen**
###### Motivation for this change

Version bump, closes #157574. Also re-enables builds on non-x86.
@ericdallo @thiagokokada 

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [X] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by willcohen** [comment]
Result of `nixpkgs-review pr 157723` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>clojure-lsp</li>
    <li>vscode-extensions.betterthantomorrow.calva</li>
  </ul>
</details>

File: pkgs/development/tools/misc/clojure-lsp/default.nix [hunk 1]
```diff
@@ -2,18 +2,18 @@
...
+    sha256 = "sha256-VyDnDlK40Sj/0cethclnRlKc5tOentAEUzjDOqbItpo=";
   };
 
   jar = fetchurl {
     url = "https://github.com/clojure-lsp/clojure-lsp/releases/download/${version}/clojure-lsp-standalone.jar";
-    sha256 = "sha256-le0+ZmGyhM/UfGUV05FHyx5PlARxL/T2aC8UhiPYjxw";
+    sha256 = "sha256-gIG8sjf55aRo4xI/RFrxhzLSD6RHzn/YGG9+gWrXcgE=";
   };
 
   # https://github.com/clojure-lsp/clojure-lsp/blob/2022.01.22-01.31.09/cli/graalvm/native-unix-compile.sh#L18-L27
```
Comment (ericdallo) [inline pkgs/development/tools/misc/clojure-lsp/default.nix:19]: We don't need this `preBuild` anymore

Comment (willcohen) [reply pkgs/development/tools/misc/clojure-lsp/default.nix:19]: Of course. Checking locally and will force-push if it works!

**Comment by willcohen** [comment]
Result of `nixpkgs-review pr 157723` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>clojure-lsp</li>
    <li>vscode-extensions.betterthantomorrow.calva</li>
  </ul>
</details>

**Review by ericdallo — APPROVED** [review]

**Comment by thiagokokada** [comment]
Result of `nixpkgs-review pr 157723` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>clojure-lsp</li>
    <li>vscode-extensions.betterthantomorrow.calva</li>
  </ul>
</details>

**Review by thiagokokada — APPROVED** [review]

**Pull request merged**